File: VB/WinSolution.Module.Win/TreeListAppearanceControllerEx.vb
Imports DevExpress.ExpressApp.Editors
Imports DevExpress.ExpressApp.TreeListEditors.Win
Imports DevExpress.ExpressApp.Win.Controls
Imports DevExpress.ExpressApp.Win.Editors
Imports DevExpress.XtraTreeList

Namespace WinSolution.Module.Win
    Public Class TreeListAppearanceControllerEx
        Inherits TreeListAppearanceController

        Protected Overrides Sub OnTreeListChanged()
            MyBase.OnTreeListChanged()
            If MyBase.Active.ResultValue AndAlso MyBase.View IsNot Nothing AndAlso MyBase.View.Editor IsNot Nothing AndAlso TypeOf MyBase.View.Editor Is TreeListEditor AndAlso CType(MyBase.View.Editor, TreeListEditor).TreeList IsNot Nothing Then
                AddHandler CType(View.Editor, TreeListEditor).TreeList.ShowingEditor, AddressOf control_ShowingEditor
            End If
        End Sub
        Private Sub control_ShowingEditor(ByVal sender As Object, ByVal e As System.ComponentModel.CancelEventArgs)
            Dim tl As TreeList = DirectCast(sender, TreeList)
            Dim node As ObjectTreeListNode = TryCast(tl.FocusedNode, ObjectTreeListNode)
            If node Is Nothing Then
                Return
            End If
            Me.OnCustomizeAppearance(New CustomizeAppearanceEventArgs(tl.FocusedColumn.FieldName, "ViewItem", New GridViewCancelEventArgsAppearanceAdapter(Nothing, e), node.Object, Nothing))
        End Sub
    End Class
End Namespace

File: CS/WinSolution.Module.Win/TreeListAppearanceControllerEx.cs
using DevExpress.ExpressApp.Editors;
using DevExpress.ExpressApp.TreeListEditors.Win;
using DevExpress.ExpressApp.Win.Controls;
using DevExpress.ExpressApp.Win.Editors;
using DevExpress.XtraTreeList;

namespace WinSolution.Module.Win {
    public class TreeListAppearanceControllerEx : TreeListAppearanceController {
        protected override void OnTreeListChanged() {
            base.OnTreeListChanged();
            if (base.Active.ResultValue && base.View != null && base.View.Editor != null && base.View.Editor is TreeListEditor && ((TreeListEditor)base.View.Editor).TreeList != null) {
                ((TreeListEditor)base.View.Editor).TreeList.ShowingEditor += new System.ComponentModel.CancelEventHandler(control_ShowingEditor);
            }
        }
        void control_ShowingEditor(object sender, System.ComponentModel.CancelEventArgs e) {
            TreeList tl = (TreeList)sender;
            ObjectTreeListNode node = tl.FocusedNode as ObjectTreeListNode;
            if (node == null) return;
            this.OnCustomizeAppearance(new CustomizeAppearanceEventArgs(tl.FocusedColumn.FieldName, "ViewItem", new GridViewCancelEventArgsAppearanceAdapter(null, e), node.Object, null));
        }
    }
}
